### PR TITLE
Module2

### PR DESCRIPTION
**Pull request by antonpus**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
-import {User, Product, config} from './exports';
+import {DirWatcher, Importer} from './exports';
+import * as path from 'path';
+import * as events from 'events';
 
-console.log(config.name);
+const getDirPath = () => `${path.dirname(require.main.filename)}/../data`;
```
Comment (vinfinit) [inline src/app.js]: Clearer way 
```javascript
const getDirPath = () => `${__dirname}/../data`;
```

File: src/dirwatcher.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+const readDirAsync = util.promisify(fs.readdir);
+
+const watchNewFiles = function watchNewFiles(dirPath, fileNames) {
+  fileNames
+  .filter(fileName => isNewFile.call(this, fileName))
+  .forEach(fileName => {
+    this.eventEmitter.emit('dirwatcher:changed', path.join(dirPath, fileName));
+    this.watchedFiles.push(fileName);
```
Comment (vinfinit) [inline src/dirwatcher.js:35]: Need to be fixed:
Don't react when a user removes files. As a result, when you remove a file and then add a file with the same file name there is no reaction.
Expected: when a user removes files, the files should be excluded from watchedFiles array.

File: src/dirwatcher.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        .then(fileNames => watchNewFiles.call(this, dirPath, fileNames))
+        .catch(err => console.error(`An error happened while trying to read
+           ${dirPath}. ${err}`))
+    }, delay)
+  }
+}
+
+const readDirAsync = util.promisify(fs.readdir);
+
+const watchNewFiles = function watchNewFiles(dirPath, fileNames) {
```
Comment (vinfinit) [inline src/dirwatcher.js:30]: Private functions are cool.

**Review by vinfinit — CHANGES_REQUESTED** [review]
Need to be fixed: 1.
Please, fix the bug with files removing.

File: src/dirwatcher.js (hunk 2 above)
Comment (antonpus) [reply src/dirwatcher.js:35]: Done.

File: src/app.js (hunk 1 above)
Comment (antonpus) [reply src/app.js]: thnx, fixed.

**Review by vinfinit — APPROVED** [review]

**Pull request merged**